core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -10512,41 +10515,47 @@ private void checkCustomColumnResolving(String table) {
         + "key => descriptor(productid),\n"
         + "size => interval '1' hour))")
         .ok();
-    // negative tests.
     sql("select * from table(\n"
         + "session(\n"
-        + "^\"data\"^ => table orders,\n"
+        + "data => table orders,\n"
         + "timecol => descriptor(rowtime),\n"
-        + "key => descriptor(productid),\n"
         + "size => interval '1' hour))")
-        .fails("Param 'data' not found in function 'SESSION'; did you mean 'DATA'\\?");
+        .ok();
+    // negative tests.
     sql("select * from table(\n"
         + "^session(\n"
         + "data => table orders,\n"
         + "key => descriptor(productid),\n"
         + "size => interval '1' hour)^)")
-        .fails("Invalid number of arguments to function 'SESSION'. Was expecting 4 arguments");
+        .fails("Cannot apply 'SESSION' to arguments of type 'SESSION\\(<RECORDTYPE\\(TIMESTAMP\\("
+            + "0\\) ROWTIME, INTEGER PRODUCTID, INTEGER ORDERID\\)>, <COLUMN_LIST>, "
+            + "<INTERVAL HOUR>\\)'. Supported form\\(s\\): SESSION\\(TABLE table_name, DESCRIPTOR\\("
+            + "timecol\\), DESCRIPTOR\\(key\\) optional, datetime interval\\)");
     sql("select * from table(\n"
-        + "^session(table orders, descriptor(rowtime), interval '2' hour)^)")
-        .fails("Invalid number of arguments to function 'SESSION'. Was expecting 4 arguments");
+        + "session(\n"
+        + "^\"data\"^ => table orders,\n"
+        + "timecol => descriptor(rowtime),\n"
+        + "key => descriptor(productid),\n"
```

Comment:
One thing I am not sure is, does current Calcite named argument has an ordering requirement? Will Calcite accept if key/timecol is reordered?

Reply:
Yes, Calcite will permute the operands before checking if it's a `SqlArgumentAssignmentOperator`, so here is ok.

Reply:
Thanks for the clarification.